google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/QueryRequest.java

```diff
@@ -102,6 +106,33 @@ public Builder setQuery(String query) {
     }
 
     /**
+     * Adds a QueryParameter to the list of query parameters. See
+     * {@link #setQueryParameters(Iterable)} for more details on the input requirements.
+     */
+    public Builder addQueryParameter(QueryParameter queryParameter) {
+      if (queryParameters == null) {
+        queryParameters = Lists.newArrayList();
+      }
+      queryParameters.add(checkNotNull(queryParameter));
```

Comment:
Nit, but should we have an empty list if the given parameter was null (and therefore rejected)?
Looking at the code I see that you handle it later, so a very minor nit which could be ignored.